### PR TITLE
DEVO-635: Setting up pipeline to run unit tests on javaclient api

### PR DESCRIPTION
**Pull request by SameeraPriyathamTadikonda**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          export PATH=$GRADLE_USER_HOME:$JAVA_HOME/bin:$PATH
+          cd java-client-api
+          ./gradlew mlDeploy
+        '''
+        sh label:'test', script: '''#!/bin/bash
+          export JAVA_HOME=$JAVA_HOME_DIR
+          export GRADLE_USER_HOME=$WORKSPACE/$GRADLE_DIR
+          export PATH=$GRADLE_USER_HOME:$JAVA_HOME/bin:$PATH
+          cd java-client-api
+          ./gradlew test  || true
```
Comment (rjrudin) [inline Jenkinsfile:31]: @SameeraPriyathamTadikonda I think this should just be marklogic-client-api:test - I don't think we need to run the tests for the other two projects yet, but maybe later.

**Review by rjrudin — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (SameeraPriyathamTadikonda) [reply Jenkinsfile]: I changed to run functional tests to run after the PR merges.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  environment{
+    JAVA_HOME_DIR="/home/builder/java/openjdk-1.8.0-262"
+    GRADLE_DIR   =".gradle"
+    DMC_USER     = credentials('MLBUILD_USER')
+    DMC_PASSWORD = credentials('MLBUILD_PASSWORD')
+  }
+  stages{
+    stage('tests'){
+      steps{
+        copyRPM 'Release','10.0-9.4'
```
Comment (rjrudin) [inline Jenkinsfile:17]: @anu3990  @llinggit  Do you think we should use the latest ML release or the latest nightly build?

Comment (llinggit) [reply Jenkinsfile:17]: sorry I missed the context here. We've already got a Jenkins job to run java unit tests against nightly build, right?

Comment (rjrudin) [reply Jenkinsfile:17]: This will run against submitted PRs before they're merged

Comment (SameeraPriyathamTadikonda) [reply Jenkinsfile:17]: We can also run the tests on a nightly server after merging the PR to develop.

Comment (llinggit) [reply Jenkinsfile:17]: nightly build is better I think.

**Review by rjrudin — APPROVED** [review]
LGTM, and I'm good with it going against the nightly instead

**Pull request merged**